gph/src/ripser.h

```diff
@@ -280,24 +280,30 @@ struct compressed_distance_matrix {
 
     compressed_distance_matrix(std::vector<value_t>&& _distances)
         : distances(std::move(_distances)),
-          rows((1 + std::sqrt(1 + 8 * distances.size())) / 2)
+          rows((-1 + std::sqrt(1 + 8 * distances.size())) / 2)
     {
-        assert(distances.size() == size() * (size() - 1) / 2);
+        assert(distances.size() == size() * (size() + 1) / 2);
         init_rows();
     }
 
     template <typename DistanceMatrix>
     compressed_distance_matrix(const DistanceMatrix& mat)
-        : distances(mat.size() * (mat.size() - 1) / 2), rows(mat.size())
+        : distances(mat.size() * (mat.size() + 1) / 2), rows(mat.size())
     {
         init_rows();
 
-        for (size_t i = 1; i < size(); ++i)
-            for (size_t j = 0; j < i; ++j)
+        for (size_t i = 0; i < size(); ++i)
+            for (size_t j = 0; j <= i; ++j)
                 rows[i][j] = mat(i, j);
     }
 
-    value_t operator()(const index_t i, const index_t j) const;
+    inline value_t mat_tri_val(const index_t min, const index_t max) const;
+
+    value_t operator()(const index_t i, const index_t j) const
+    {
+        const auto& p = std::minmax(i, j);
+        return mat_tri_val(p.first, p.second);
```

Comment:
Unfortunately, this we need this syntax because our compiler minimum standard supported version is `c++14`. If the minimal supported version would be `c++17`, the we could use the following syntax:

```cpp
if constexpr(std::is_same_v<Layout, compressed_lower_distance_matrix>)
    return rows[p.second][p.first];
else
    return rows[p.first][p.second];
```

This is IMO, much cleaner code, but I wouldn't make `c++17` the minimum requirement just for this feature. This feature is also interesting, because it is evaluated at compile time, that should produce similar code to the current implementation

Reply:
Same suggestion about an inline comment as above.

Reply:
Done for both !
Feel free to modify and integrate if necessary 👍🏼 
You can then resolve conversations

Reply:
Awesome, thank you!